src/crypto/aead/chacha20poly1305.rs

```diff
@@ -0,0 +1,187 @@
+//! The original NaCl variant of the Chacha20-Poly1305 construction,
+//! an algorithm for Authenticated Encryption with Additional Data
+//! (AEAD).
+
```

Comment:
The documentation here is a bit misleading. NaCl doesn't contain Chacha20-Poly1305.